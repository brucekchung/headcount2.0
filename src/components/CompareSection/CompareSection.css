
.comparison {
  display: flex;
  justify-content: space-around;
}

.comparison .card {
  margin: auto;
}