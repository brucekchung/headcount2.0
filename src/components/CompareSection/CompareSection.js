import React from 'react';
import Card from '../card/Card';
import PropTypes from 'prop-types';

const CompareSection = ({ data, calculateAverage }) => {
  const [a, b] = data;
  let averages;

  if (data.length === 2) {
    averages = calculateAverage(a, b);
  }

  const compareCard = () => {
    return(
      <div className="card">
        <h3>{ a.location }: { averages[a.location] } </h3>
        <h2>{ averages.compared }</h2>
        <h3>{ b.location }: { averages[b.location] }</h3>
      </div>
    )
  }

  
  return (
    <div>
      {
        data.length === 1 &&
        <Card { ...a } className="card selected" />
      }
      {
        data.length === 2 &&
        <div>
          <Card { ...a } className="card selected" />
          { 
            compareCard() 
          }
          <Card { ...b } className="card selected" />
        </div>
      }
    </div>
  )
}

export default CompareSection