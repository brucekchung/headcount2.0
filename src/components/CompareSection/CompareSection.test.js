import React from 'react'
import CompareSection from './CompareSection'
import {shallow} from 'enzyme'
import DistrictRepository from '../../helper.js';
import kinderData from '../../data/kindergartners_in_full_day_program.js';

import {configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe('CompareSection', () => {
  let wrapper
  const district = new DistrictRepository(kinderData)
  const calculateAverage = (a, b) => district.compareDistrictAverages(a.location, b.location)
  const compare = [{
    "location": "ColoraDo",
    "data": {2004: 0.756, 2005: 0.658, 2006: 0.253},
    "id": 1
  },
  {
    "location": "ACADEMY 20",
    "data": {2004: 0.465, 2005: 0.758, 2006: 0.856},
    "id": 2
  }]

  beforeEach(() => {
    wrapper = shallow(<CompareSection data={compare}
                                      calculateAverage={calculateAverage} />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render two district cards and a comparison card', () => {
    expect(wrapper.find('Card').length).toEqual(2)
    expect(wrapper.find('.center').length).toEqual(1)
  })

  it('should display the average of both districts', () => {
    expect(wrapper.find('h2').text()).toEqual('1.3');
  })

  it('should render a single card if there is only one selected', () => {
    const selected=[{
      "location": "ColoraDo",
      "data": {2004: 0.756, 2005: 0.658, 2006: 0.253},
      "id": 1
    }]
    wrapper = shallow(<CompareSection data={selected}
                                      calculateAverage={calculateAverage} />)

    expect(wrapper.find('Card').length).toEqual(1)
  })

})
