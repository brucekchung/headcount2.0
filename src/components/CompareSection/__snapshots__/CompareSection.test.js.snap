// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareSection should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareSection
    calculateAverage={[Function]}
    data={
        Array [
            Object {
              "data": Object {
                "2004": 0.756,
                "2005": 0.658,
                "2006": 0.253,
              },
              "id": 1,
              "location": "ColoraDo",
            },
            Object {
              "data": Object {
                "2004": 0.465,
                "2005": 0.758,
                "2006": 0.856,
              },
              "id": 2,
              "location": "ACADEMY 20",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="render-multiple"
>
        <Card
                className="card selected"
                data={
                        Object {
                                "2004": 0.756,
                                "2005": 0.658,
                                "2006": 0.253,
                              }
                }
                id={1}
                location="ColoraDo"
        />
        <div
                className="card center"
        >
                <h3>
                        ACADEMY 20
                        : 
                         
                </h3>
                <h2>
                        1.3
                </h2>
                <h3>
                        ACADEMY 20
                        : 
                        0.407
                </h3>
        </div>
        <Card
                className="card selected"
                data={
                        Object {
                                "2004": 0.465,
                                "2005": 0.758,
                                "2006": 0.856,
                              }
                }
                id={2}
                location="ACADEMY 20"
        />
</div>,
      "className": "comparison",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            className="card selected"
            data={
                        Object {
                                    "2004": 0.756,
                                    "2005": 0.658,
                                    "2006": 0.253,
                                  }
            }
            id={1}
            location="ColoraDo"
/>,
          <div
            className="card center"
>
            <h3>
                        ACADEMY 20
                        : 
                         
            </h3>
            <h2>
                        1.3
            </h2>
            <h3>
                        ACADEMY 20
                        : 
                        0.407
            </h3>
</div>,
          <Card
            className="card selected"
            data={
                        Object {
                                    "2004": 0.465,
                                    "2005": 0.758,
                                    "2006": 0.856,
                                  }
            }
            id={2}
            location="ACADEMY 20"
/>,
        ],
        "className": "render-multiple",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "card selected",
            "data": Object {
              "2004": 0.756,
              "2005": 0.658,
              "2006": 0.253,
            },
            "id": 1,
            "location": "ColoraDo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                ACADEMY 20
                : 
                 
</h3>,
              <h2>
                1.3
</h2>,
              <h3>
                ACADEMY 20
                : 
                0.407
</h3>,
            ],
            "className": "card center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "ACADEMY 20",
                  ": ",
                  undefined,
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "ACADEMY 20",
                ": ",
                undefined,
                " ",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1.3,
              },
              "ref": null,
              "rendered": 1.3,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "ACADEMY 20",
                  ": ",
                  0.407,
                ],
              },
              "ref": null,
              "rendered": Array [
                "ACADEMY 20",
                ": ",
                0.407,
              ],
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "card selected",
            "data": Object {
              "2004": 0.465,
              "2005": 0.758,
              "2006": 0.856,
            },
            "id": 2,
            "location": "ACADEMY 20",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="render-multiple"
>
          <Card
                    className="card selected"
                    data={
                              Object {
                                        "2004": 0.756,
                                        "2005": 0.658,
                                        "2006": 0.253,
                                      }
                    }
                    id={1}
                    location="ColoraDo"
          />
          <div
                    className="card center"
          >
                    <h3>
                              ACADEMY 20
                              : 
                               
                    </h3>
                    <h2>
                              1.3
                    </h2>
                    <h3>
                              ACADEMY 20
                              : 
                              0.407
                    </h3>
          </div>
          <Card
                    className="card selected"
                    data={
                              Object {
                                        "2004": 0.465,
                                        "2005": 0.758,
                                        "2006": 0.856,
                                      }
                    }
                    id={2}
                    location="ACADEMY 20"
          />
</div>,
        "className": "comparison",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              className="card selected"
              data={
                            Object {
                                          "2004": 0.756,
                                          "2005": 0.658,
                                          "2006": 0.253,
                                        }
              }
              id={1}
              location="ColoraDo"
/>,
            <div
              className="card center"
>
              <h3>
                            ACADEMY 20
                            : 
                             
              </h3>
              <h2>
                            1.3
              </h2>
              <h3>
                            ACADEMY 20
                            : 
                            0.407
              </h3>
</div>,
            <Card
              className="card selected"
              data={
                            Object {
                                          "2004": 0.465,
                                          "2005": 0.758,
                                          "2006": 0.856,
                                        }
              }
              id={2}
              location="ACADEMY 20"
/>,
          ],
          "className": "render-multiple",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "card selected",
              "data": Object {
                "2004": 0.756,
                "2005": 0.658,
                "2006": 0.253,
              },
              "id": 1,
              "location": "ColoraDo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  ACADEMY 20
                  : 
                   
</h3>,
                <h2>
                  1.3
</h2>,
                <h3>
                  ACADEMY 20
                  : 
                  0.407
</h3>,
              ],
              "className": "card center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "ACADEMY 20",
                    ": ",
                    undefined,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "ACADEMY 20",
                  ": ",
                  undefined,
                  " ",
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1.3,
                },
                "ref": null,
                "rendered": 1.3,
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "ACADEMY 20",
                    ": ",
                    0.407,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "ACADEMY 20",
                  ": ",
                  0.407,
                ],
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "card selected",
              "data": Object {
                "2004": 0.465,
                "2005": 0.758,
                "2006": 0.856,
              },
              "id": 2,
              "location": "ACADEMY 20",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
